src/react/deckgl.js

```diff
@@ -147,6 +166,52 @@ export default class DeckGL extends React.Component {
     }
   }
 
+  _onDragEvent(event, explicitType) {
+    // use offsetX|Y for relative position to the container, drop event if falsy
+    if (!event || !event.event || !Number.isFinite(event.event.offsetX)) {
+      return;
+    }
+    const {event: {offsetX: x, offsetY: y}} = event;
+    const type = typeof explicitType === 'string' ? explicitType : event.event.type;
+    let mode;
+    let layerEventHandler;
+    switch (type) {
+    case 'mousedown':
```

Comment:
This PR doesn't as far as supporting touch events. Since we're planning to refactor event handling out anyway, I'm deferring touch event handling for that refactor and focusing only on drag events here (for the graph-layer needs). I'll remove them from `MOTION_EVENTS` for consistency.